test/module/irohad/multi_sig_transactions/mst_processor_test.cpp

```diff
@@ -0,0 +1,262 @@
+/**
+ * Copyright Soramitsu Co., Ltd. 2017 All Rights Reserved.
+ * http://soramitsu.co.jp
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <gtest/gtest.h>
+#include "framework/test_subscriber.hpp"
+#include "logger/logger.hpp"
+#include "module/irohad/multi_sig_transactions/mst_mocks.hpp"
+#include "module/irohad/multi_sig_transactions/mst_test_helpers.hpp"
+#include "multi_sig_transactions/mst_processor_impl.hpp"
+#include "multi_sig_transactions/storage/mst_storage_impl.hpp"
+
+auto log_ = logger::log("MstProcessorTest");
+
+using namespace std;
+using namespace iroha;
+using namespace iroha::model;
+using namespace framework::test_subscriber;
+
+using testing::Return;
+using testing::_;
+
+class TestCompleter : public Completer {
+  bool operator()(const DataType transaction) const override {
+    return transaction->signatures.size() >= transaction->quorum;
+  }
+
+  bool operator()(const DataType &tx, const TimeType &time) const override {
+    return tx->created_ts < time;
+  }
+};
+
+class MstProcessorTest : public testing::Test {
+ public:
+  // --------------------------------| fields |---------------------------------
+
+  Peer own_peer = makePeer("iam", "1");
+  /// propagation subject, useful for propagation control
+  rxcpp::subjects::subject<PropagationStrategy::PropagationData>
+      propagation_subject;
+  /// use effective implementation of storage
+  shared_ptr<MstStorage> storage;
+  shared_ptr<FairMstProcessor> mst_processor;
+
+  // ---------------------------------| mocks |---------------------------------
+
+  shared_ptr<MockMstTransport> transport;
+  shared_ptr<MockPropagationStrategy> propagation_strategy;
+  shared_ptr<MockTimeProvider> time_provider;
+
+ protected:
+  void SetUp() override {
+    transport = make_shared<MockMstTransport>();
+    storage = make_shared<MstStorageStateImpl>(own_peer,
+                                               make_shared<TestCompleter>());
+
+    propagation_strategy = make_shared<MockPropagationStrategy>();
+    EXPECT_CALL(*propagation_strategy, emitter())
```

Comment:
is it a good idea to place EXPECT_CALL in the fixture's `SetUp`?

Reply:
Sure, if this behaviour suitable for all test scenarios.

Reply:
I mean does this calls performed in the SetUp?
E.g if i write a new test that calls this functions (could i, btw?) it would be automatically be broken?

Reply:
You may add explicit `EXPECT_CALL` in your test. Or inherit from the fixture and override  `SetUp`.